port_ocean/config/base.py

```diff
@@ -58,16 +59,43 @@ def decamelize_object(obj: Any) -> Any:
         return obj
 
 
-def load_providers(settings: "BaseOceanSettings", base_path: str) -> dict[str, Any]:
+def parse_config(
+    config: dict[str, Any], existing_data: dict[str, Any]
+) -> dict[str, Any]:
+    """
+    Normalizing the config yaml file to work with snake_case and getting only the data that is missing for the settings
+    """
+    for key, value in config.items():
```

Comment:
i think the loop can be a tiny bit simpler if you'll create a variable at the start `newValue = value` and in the `if`s set newValue's value, then at the end of the for do the decamelized-key and set the existing data value

also, consider if you want to keep the camelized key on the existing data or not

Reply:
i cant do the firs suggestion because there is a scenario i will want to skip that key